### PR TITLE
test the built image against postgresql

### PR DESCRIPTION
**Pull request by klaernie**
this is a verbatim copy of the mariadb test against, testing against a
16.3 postgresql image based on alpine (16.4 is still building in the
docker library).

### CODE REVIEW

**Comment by klaernie** [comment]
FYI: I took the liberty of renaming the mariadb test directory for clarity when my brain visits these directories late in the night some day..

**Comment by klaernie** [comment]
another addendum: the last commit splits building the image and pushing it to the registry into two separate parts, so the pushing half can be skipped in a pull request triggered run - with the result that the build and test phases could complete and test the code without needing access to ghcr.io/tigattack

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -73,31 +70,73 @@ jobs:
...
+    - name: Load image into runner docker daemon
+      run: |
+        docker load --input /tmp/${{ env.IMAGE_NAME }}.tar
+
+    - name: Test run Docker Compose with a mariadb
       uses: ElParking/docker-compose-action@1.4.5
-      id: test
+      id: test_mariadb
       with:
         compose-file: .github/test/docker-compose.yml
```
Comment (tigattack) [inline .github/workflows/build.yml:81]:
```suggestion
        compose-file: .github/test-mariadb/docker-compose.yml
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -73,31 +70,73 @@ jobs:
...
         down-flags: "--volumes"
       env:
-        IMAGE: ${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.IMAGE_NAME }}@${{ steps.build.outputs.digest }}
+        IMAGE: "${{ steps.build.outputs.imageid }}"
         IMAGE_PLATFORM: ${{ matrix.platform }}
         PWD: ${{ github.workspace }}/.github/test
```
Comment (tigattack) [inline .github/workflows/build.yml:86]:
```suggestion
        PWD: ${{ github.workspace }}/.github/test-mariadb
```

**Review by tigattack — CHANGES_REQUESTED** [review]
Hey, this is great, thank you!

I've suggested a couple of fixes for the old mariadb test path.

Thanks for working on the fix for PR runs too, I'd been meaning to take a look at that. Without having checked yet, I'm wondering if the "Build and push image" step, now moved to the end of the `build` job, couple be optimised at all. Maybe the already built image could be loaded and pushed without building it again before pushing? I imagine it'll use the cache from the first push (though I've never tested using a cache created earlier on in the same job myself), but it still feels slightly inefficient to build twice.

**Comment by klaernie** [comment]
Oh, now the test revealed another flaw - the postgres and mariadb servers need to use different ports. I'll fix that once I have access to something better than my phone..

**Comment by klaernie** [comment]
Now it finally runs properly.. but that docker compose action bug needs a deeper look from me later on..

**Review by tigattack — APPROVED** [review]
Cheers, looks good!

**Pull request merged**